lib/steps.js

```diff
@@ -576,6 +577,108 @@ function detectCompendium(passon) {
+                let message = 'No id found in compendium detection file';
+                let error = new Error(message);
+                error.msg = message;
+                error.status = 400;
+                reject(error);
+            } else {
+                // Validate compendium id
+                let idRegex = new RegExp(config.bagtainer.id_regex);
+                if (idRegex.test(compendiumID)) {
+
+                    // Check if id found in compendium file already exists
+                    Compendium.findOne({ id: compendiumID }).select('id user').exec((err, compendium) => {
+                        // eslint-disable-next-line no-eq-null, eqeqeq
+                        if (err) {
+                            debug('[%s] Error querying compendium with id %s', passon.id, compendiumID);
+                            err.message = 'Error querying compendium';
+                            err.status = 400;
+                            reject(err);
+                        }
+                        if (compendium === null) {
+                            debug('[%s] Assigned new ID %s to compendium', passon.id, compendiumID);
+                            passon.id = compendiumID;
+                            fulfill(passon);
+                        } else {
+                            debug('[%s] Compendium with ID %s already exists; user %s', passon.id, compendiumID, compendium.user);
+                            let message = 'Error fetching ID from compendium, ID already exists';
+                            let error = new Error(message);
+                            error.msg = message;
+                            error.status = 400;
+                            reject(error);
+                        }
+                    });
+                } else {
+                    debug('[%s] Invalid ID %s specified for compendium in %s', passon.id, compendiumID, passon.detectionFile);
+                    let message = 'Invalid id found in compendium detection file';
+                    let error = new Error(message);
+                    error.msg = message;
+                    error.status = 400;
+                    reject(error);
+                }
+            }
+        } else {
+            debug('[%s] Not a compendium, keeping generated ID', passon.id);
+            fulfill(passon);
+        }
+    });
+}
+
+function moveCompendiumFiles(passon) {
+    return new Promise((fulfill, reject) => {
+
+        if (passon.isCompendium){
+            let updatedPath = path.join(config.fs.compendium, passon.id);
+            debug('[%s] Copying compendium files from %s to %s due to ID specified in compendium file', passon.id, passon.compendium_path, updatedPath);
+
+            let cmd = 'mv ' + passon.compendium_path + ' ' + updatedPath;
+            exec(cmd, (error, stdout, stderr) => {
+                if (error || stderr) {
+                    debug(error, stderr, stdout);
```

Comment:
Please add a short message to this debug statement. Printing all outputs is good, a short text helps identifying where the error happens.